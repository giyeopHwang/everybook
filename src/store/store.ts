import { configureStore } from '@reduxjs/toolkit';
import { RootState } from '@/data/types';
import displayOptionsReducer from './display-options/display-options-slice';

export default configureStore<RootState>({
  reducer: {
    displayOptions: displayOptionsReducer,
  },
});
