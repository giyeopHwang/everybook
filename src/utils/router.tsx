import { createBrowserRouter } from 'react-router-dom';

import Front from '@/components/front/Front.component';
import DashboardLayout from '@/components/dashboard/dashboard-layout/DashboardLayout.compoent';

export const router = createBrowserRouter([
  {
    path: '/',
    element: <Front />,
  },
  {
    path: 'dashboard',
    element: <DashboardLayout />,
    children: [
      {
        path: '',
        element: <div>가계부</div>,
      },
      {
        path: 'report',
        element: <div>보고서</div>,
      },
      {
        path: 'budget',
        element: <div>예산 관리</div>,
      },
      {
        path: 'account',
        element: <div>분류 관리</div>,
      },
    ],
  },
]);
