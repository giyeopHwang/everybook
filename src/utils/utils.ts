import dayjs from 'dayjs';

export const displayAmount = (
  amount: number,
  showAmount: boolean = true,
  placeholder: string = '*********'
) => {
  return showAmount ? amount.toLocaleString() : placeholder;
};

export const formatDate = (date: Date = new Date()) => {
  const day = ['일', '월', '화', '수', '목', '금', '토'][dayjs(date).day()];
  return dayjs(date).format(`YYYY년 M월 D일 (${day})`);
};
