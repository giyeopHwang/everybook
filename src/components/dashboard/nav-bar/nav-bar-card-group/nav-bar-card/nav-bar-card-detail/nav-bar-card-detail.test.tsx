import { renderWithWrapper } from '@/test/test.utils';
import { RootState } from '@/store/store';
import { mockPlus } from '@/test/test.mock';
import NavBarCardDetails from './nav-bar-card-detail.component';

describe('NavBarCardDetails', () => {
  it('renders NavBarCardDetails component', () => {
    const navBarCardDetails = renderWithWrapper(
      <NavBarCardDetails details={mockPlus.details} />
    );

    expect(navBarCardDetails).toMatchSnapshot();
  });

  it('renders with hidden amount', () => {
    const initialState: Partial<RootState> = {
      displayOptions: {
        showAmount: false,
        showNavBar: true,
      },
    };

    const navBarCard = renderWithWrapper(
      <NavBarCardDetails details={mockPlus.details} />,
      { preloadedState: initialState }
    );

    expect(navBarCard).toMatchSnapshot();
  });
});
