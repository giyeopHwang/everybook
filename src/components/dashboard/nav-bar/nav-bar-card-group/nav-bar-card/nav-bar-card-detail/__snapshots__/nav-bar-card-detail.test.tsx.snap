// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBarCardDetails > renders NavBarCardDetails component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="css-1r2jysi MuiBox-root css-0"
      >
        <div
          class="css-uf1ume MuiBox-root css-0"
        >
          <div
            class="css-44nm6q MuiBox-root css-0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-r6guz9-MuiSvgIcon-root"
              data-testid="SubdirectoryArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m19 15-6 6-1.42-1.42L15.17 16H4V4h2v10h9.17l-3.59-3.58L13 9l6 6z"
              />
            </svg>
            <span
              class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
            >
              PLUS_ITEM_1
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            100
          </span>
        </div>
        <div
          class="css-uf1ume MuiBox-root css-0"
        >
          <div
            class="css-44nm6q MuiBox-root css-0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-r6guz9-MuiSvgIcon-root"
              data-testid="SubdirectoryArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m19 15-6 6-1.42-1.42L15.17 16H4V4h2v10h9.17l-3.59-3.58L13 9l6 6z"
              />
            </svg>
            <span
              class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
            >
              PLUS_ITEM_2
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            200
          </span>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="css-1r2jysi MuiBox-root css-0"
    >
      <div
        class="css-uf1ume MuiBox-root css-0"
      >
        <div
          class="css-44nm6q MuiBox-root css-0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-r6guz9-MuiSvgIcon-root"
            data-testid="SubdirectoryArrowRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m19 15-6 6-1.42-1.42L15.17 16H4V4h2v10h9.17l-3.59-3.58L13 9l6 6z"
            />
          </svg>
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            PLUS_ITEM_1
          </span>
        </div>
        <span
          class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
        >
          100
        </span>
      </div>
      <div
        class="css-uf1ume MuiBox-root css-0"
      >
        <div
          class="css-44nm6q MuiBox-root css-0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-r6guz9-MuiSvgIcon-root"
            data-testid="SubdirectoryArrowRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m19 15-6 6-1.42-1.42L15.17 16H4V4h2v10h9.17l-3.59-3.58L13 9l6 6z"
            />
          </svg>
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            PLUS_ITEM_2
          </span>
        </div>
        <span
          class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
        >
          200
        </span>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`NavBarCardDetails > renders with hidden amount 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="css-1r2jysi MuiBox-root css-0"
      >
        <div
          class="css-uf1ume MuiBox-root css-0"
        >
          <div
            class="css-44nm6q MuiBox-root css-0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-r6guz9-MuiSvgIcon-root"
              data-testid="SubdirectoryArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m19 15-6 6-1.42-1.42L15.17 16H4V4h2v10h9.17l-3.59-3.58L13 9l6 6z"
              />
            </svg>
            <span
              class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
            >
              PLUS_ITEM_1
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            *********
          </span>
        </div>
        <div
          class="css-uf1ume MuiBox-root css-0"
        >
          <div
            class="css-44nm6q MuiBox-root css-0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-r6guz9-MuiSvgIcon-root"
              data-testid="SubdirectoryArrowRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m19 15-6 6-1.42-1.42L15.17 16H4V4h2v10h9.17l-3.59-3.58L13 9l6 6z"
              />
            </svg>
            <span
              class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
            >
              PLUS_ITEM_2
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            *********
          </span>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="css-1r2jysi MuiBox-root css-0"
    >
      <div
        class="css-uf1ume MuiBox-root css-0"
      >
        <div
          class="css-44nm6q MuiBox-root css-0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-r6guz9-MuiSvgIcon-root"
            data-testid="SubdirectoryArrowRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m19 15-6 6-1.42-1.42L15.17 16H4V4h2v10h9.17l-3.59-3.58L13 9l6 6z"
            />
          </svg>
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            PLUS_ITEM_1
          </span>
        </div>
        <span
          class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
        >
          *********
        </span>
      </div>
      <div
        class="css-uf1ume MuiBox-root css-0"
      >
        <div
          class="css-44nm6q MuiBox-root css-0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-r6guz9-MuiSvgIcon-root"
            data-testid="SubdirectoryArrowRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m19 15-6 6-1.42-1.42L15.17 16H4V4h2v10h9.17l-3.59-3.58L13 9l6 6z"
            />
          </svg>
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            PLUS_ITEM_2
          </span>
        </div>
        <span
          class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
        >
          *********
        </span>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
